src/util.ts

```diff
@@ -0,0 +1,87 @@
+import { Logger } from './logger';
+
+const temp = require('temp');
+const fs = require('fs');
+const logger:Logger = Logger.getInstance();
+
+
+export class TempFileHandler {
+    track: any;
+
+    create(TextEditor) {
+        let tempFile = temp.openSync({suffix: '.py'});
+        fs.writeSync(tempFile.fd, TextEditor.getText());
+        fs.closeSync(tempFile.fd);
+        return new TempFileWrapper(tempFile);
+    }
+}
+
+
+export class TempFileWrapper {
+    tempFile: any;
+    path: String;
+
+    constructor(tempFile) {
+        this.tempFile = tempFile;
+        this.path = tempFile.path;
+    }
+
+    clean() {
+        if (this.tempFile == null) {
+            return;
+        }
+        fs.unlink(this.path);
+        this.path = null;
+        this.tempFile = null;
+    }
+
+
+}
+
+
+export function canExecute(path) {
+    try {
+        fs.accessSync(path, fs.R_OK | fs.X_OK);
+        logger.log(">>> EXECUTE CHECK <<<");
+        logger.log("> Path can be executed");
+        logger.log(">>> END <<<");
+        return true;
+    }
+    catch(err) {
+        logger.log(">>> EXECUTE CHECK <<<");
+        logger.log("> Path can not be executed:");
+        logger.log(err);
+        logger.log(">>> END <<<");
+        return false;
+    }
+}
+
+
+export function canRead(path) {
+    try {
+        fs.accessSync(path, fs.R_OK);
+        logger.log(">>> READ CHECK <<<");
+        logger.log(">>> Path can be read");
+        return true;
+    }
+    catch(err) {
+        logger.log(">>> READ CHECK <<<");
+        logger.log("> Path can not be read");
+        logger.log(err);
+        return false;
+    }
+}
+
+
+export function calculateUnderlineRange(line, rowNumber, colNumber, config) {
```

Comment:
You might look into the [`rangeFromLineNumber()` function](https://github.com/steelbrain/atom-linter/blob/v6.0.0/src/index.js#L15) in [atom-linter](https://github.com/steelbrain/atom-linter), assuming you can get a `TextEditor` instance here this logic can be replaced entirely with it.

One of the primary advantages it has over this is it will automatically highlight the entire word from a specified column instead of just guessing at a length like you are doing here.